pkg/agent/handler/agentresource.go

```diff
@@ -38,5 +46,16 @@ func (h *agentResourceHandler) Handle(ctx context.Context, _ *proto.EventMeta, r
 			h.agentResourceManager.SetAgentResource(resource)
 		}
 	}
+
+	if action == proto.Event_SUBRESOURCEUPDATED && resource.Kind == traceabilityAgent {
```

Comment:
maybe we can also check that the agentstate subresource is also the subresource that changed?

perhaps do the opposite check to, fall out early if its not a subresource action or not the agentstate sub resource or not the traceability agnet kind?